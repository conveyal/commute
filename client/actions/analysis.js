import moment from 'moment'
import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addAnalysis = createAction('add analysis')
export const createAnalysis = (newAnalysis, organizationId) => {
  newAnalysis.id = uuid.v4()
  newAnalysis.lastRunDateTime = moment().unix()
  const commuters = newAnalysis.commuters
  delete newAnalysis.commuters
  return [
    addAnalysis({ analysis: newAnalysis, organizationId }),
    push(`/organizations/${organizationId}/analysis/${newAnalysis.id}`),
    makeMockTrips(newAnalysis.id, organizationId, commuters)
  ] // TODO save to server
}

const deleteLocally = createAction('delete analysis')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/analysis/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteAnalysis = (id, organizationId) => [
  deleteLocally({ analysisId: id, organizationId }),
  push(`/organizations/${organizationId}`)
]

const makeMockTrips = (analysisId, organizationId, commuters) => {
  const trips = []
  const modes = ['bike', 'car', 'transit', 'walk']
  const randRange = () => Math.random() * 0.4 + 0.8
  commuters.forEach((commuter) => {
    const curTrip = {
      commuterId: commuter.id
    }
    const tripMetricBase = Math.random()
    let bestValue = 9999999
    let bestMode
    modes.forEach((mode) => {
      // generate random trips
      const modePctMultiplier = randRange()
      const cost = tripMetricBase * 34.7 * modePctMultiplier * randRange()
      const distance = tripMetricBase * 69 * modePctMultiplier * randRange()
      const time = tripMetricBase * 7500 * modePctMultiplier * randRange()
      curTrip[mode] = {cost, distance, time}
      const tripValue = cost + time * 0.01
      if (tripValue < bestValue) bestMode = mode
    })
    curTrip.mostLikely = Object.assign({ mode: bestMode }, curTrip[bestMode])
    trips.push(curTrip)
  })
  return sendMockTrips({analysisId, organizationId, trips})
}
const sendMockTrips = createAction('receive mock calculated trips')
