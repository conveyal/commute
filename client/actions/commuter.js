import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addCommuter = createAction('add commuter')
export const createCommuter = ({commuter, groupId, organizationId}) => {
  commuter.id = uuid.v4()
  return [
    addCommuter({commuter, groupId, organizationId}),
    push(`/organizations/${organizationId}/groups/${groupId}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete commuter')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/commuter/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteCommuter = ({commuterId, groupId, organizationId}) => [
  deleteLocally({commuterId, groupId, organizationId}),
  push(`/organizations/${organizationId}/groups/${groupId}`)
]

const updateLocally = createAction('update commuter')
/* const updateOnServer = (id) =>
  serverAction({
    url: `/api/commuter/${id}`,
    options: {
      method: 'update'
    }
  }) */ // TODO update on server
export const updateCommuter = ({commuter, groupId, organizationId}) => [
  updateLocally({commuter, groupId, organizationId}),
  push(`/organizations/${organizationId}/groups/${groupId}`)
]
