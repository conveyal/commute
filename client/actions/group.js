import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addGroup = createAction('add group')
const appendCommuters = createAction('append commuters')
export const appendToGroup = (newCommuters, groupId, organizationId) => {
  return [
    appendCommuters({ commuters: newCommuters, groupId, organizationId }),
    push(`/organizations/${organizationId}/groups/${groupId}`)
  ] // TODO save to server
}
export const createGroup = (newGroup, organizationId) => {
  newGroup.id = uuid.v4()
  newGroup.allAddressesGeocoded = false
  return [
    addGroup({ group: newGroup, organizationId }),
    push(`/organizations/${organizationId}/groups/${newGroup.id}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete group')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/group/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteGroup = (id, organizationId) => [
  deleteLocally({ groupId: id, organizationId }),
  push(`/organizations/${organizationId}`)
]
