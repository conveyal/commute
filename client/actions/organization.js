import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addLocally = createAction('add organization')
export const createOrganization = (newOrganization) => {
  newOrganization.id = uuid.v4()
  newOrganization.analyses = []
  newOrganization.analysesById = {}
  newOrganization.sites = []
  newOrganization.sitesById = {}
  newOrganization.groups = []
  newOrganization.groupsById = {}
  return [
    addLocally(newOrganization),
    push(`/organizations/${newOrganization.id}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete organization')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/organization/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteOrganization = (id) => [deleteLocally(id), push('/')]

export const updateOrganization = (organization) => [
  setLocally(organization),
  push(`/organizations/${organization.id}`)
]
/* const updateOnServer = (organization) =>
  serverAction({
    url: `/api/organization/${organization.id}`,
    options: {
      body: JSON.stringify(organization),
      method: 'put'
    }
  }) */  // TODO: update on server

export const setLocally = createAction('set organization')
