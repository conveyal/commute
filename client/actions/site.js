import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'
import uuid from 'uuid'

const addSite = createAction('add site')
export const createSite = (newSite, organizationId) => {
  newSite.id = uuid.v4()
  return [
    addSite({ organizationId, site: newSite }),
    push(`/organizations/${organizationId}`)
  ] // TODO save to server
}

const deleteLocally = createAction('delete site')
/* const deleteOnServer = (id) =>
  serverAction({
    url: `/api/site/${id}`,
    options: {
      method: 'delete'
    }
  }) */ // TODO delete on server
export const deleteSite = (id, organizationId) => [
  deleteLocally({ siteId: id, organizationId }),
  push(`/organizations/${organizationId}`)
]

const updateLocally = createAction('update site')
/* const updateOnServer = (id) =>
  serverAction({
    url: `/api/site/${id}`,
    options: {
      method: 'update'
    }
  }) */ // TODO update on server
export const updateSite = (site, organizationId) => [
  updateLocally({ organizationId, site }),
  push(`/organizations/${organizationId}`)
]
