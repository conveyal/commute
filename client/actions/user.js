import {createAction} from 'redux-actions'

// login / logout
export const login = createAction('log in')
export const logout = createAction('log out')
