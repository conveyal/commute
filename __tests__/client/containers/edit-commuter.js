/* global describe, expect, it */

import {mount} from 'enzyme'
import {mountToJson} from 'enzyme-to-json'
import React from 'react'
import {Provider} from 'react-redux'

import {expectCreateCommuter, expectDeleteCommuter, expectUpdateAction} from '../../test-utils/actions'
import {makeMockStore, mockStores} from '../../test-utils/mock-data'
import Leaflet from '../../test-utils/mock-leaflet'

import EditCommuter from '../../../client/containers/edit-commuter'

describe('Container > EditCommuter', () => {
  it('Create/Edit Commuter View loads (create or edit mode)', () => {
    const mockStore = makeMockStore(mockStores.complexOrganization)

    // mount component
    mount(
      <Provider store={mockStore}>
        <EditCommuter
          params={{organizationId: '2', groupId: '1'}}
          />
      </Provider>
    , {
      attachTo: document.getElementById('test')
    })
  })

  it('Create/Edit Commuter View loads in create mode', () => {
    const mockStore = makeMockStore(mockStores.complexOrganization)

    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <EditCommuter
          params={{organizationId: '2', groupId: '1'}}
          />
      </Provider>
    , {
      attachTo: document.getElementById('test')
    })

    expect(mountToJson(tree.find('.commuter-header'))).toMatchSnapshot()
    expect(mountToJson(tree.find('.commuter-form'))).toMatchSnapshot()
    expect(mountToJson(tree.find('.commuter-submit-buttons'))).toMatchSnapshot()

    // no marker should be added initially in create mode
    expect(Leaflet.marker).not.toBeCalled()
  })

  it('Create/Edit Commuter View loads in edit mode', () => {
    const mockStore = makeMockStore(mockStores.complexOrganization)

    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <EditCommuter
          params={{organizationId: '2', groupId: '1', commuterId: '1'}}
          />
      </Provider>
    , {
      attachTo: document.getElementById('test')
    })

    expect(mountToJson(tree.find('.commuter-header'))).toMatchSnapshot()
    expect(mountToJson(tree.find('.commuter-form'))).toMatchSnapshot()
    expect(mountToJson(tree.find('.commuter-submit-buttons'))).toMatchSnapshot()

    // marker should be added initially in edit mode
    expect(Leaflet.marker).toBeCalled()
    expect(Leaflet.marker.mock.calls[0][0]).toMatchSnapshot()
  })

  it('Create commuter', () => {
    const mockStore = makeMockStore(mockStores.complexOrganization)

    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <EditCommuter
          params={{organizationId: '2', groupId: '1'}}
          />
      </Provider>
    , {
      attachTo: document.getElementById('test')
    })

    // give each text field some input
    // name
    tree.find('input').first().simulate('change', {target: {value: 'Mock Commuter'}})
    // email
    tree.find('input').at(1).simulate('change', {target: {value: 'mock@email.fake'}})
    // address
    tree.find('.form-group').find('Geocoder').props().onChange({
      type: 'Feature',
      geometry: {
        type: 'Point',
        coordinates: [
          -76.9897,
          38.89011
        ]
      },
      properties: {
        label: 'Abraham Lincoln/Emancipation Monument, Washington, USA'
      }
    })

    // submit form
    tree.find('.commuter-submit-buttons').find('button').first().simulate('click')

    // expect create action
    expectCreateCommuter(mockStore.getActions())
  })

  it('Update commuter', () => {
    const mockStore = makeMockStore(mockStores.complexOrganization)

    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <EditCommuter
          params={{organizationId: '2', groupId: '1', commuterId: '1'}}
          />
      </Provider>
    , {
      attachTo: document.getElementById('test')
    })

    // give each text field some input
    // name
    tree.find('input').first().simulate('change', {target: {value: 'Mock Commuter'}})
    // email
    tree.find('input').at(1).simulate('change', {target: {value: 'mock@email.fake'}})
    // address
    tree.find('.form-group').find('Geocoder').props().onChange({
      type: 'Feature',
      geometry: {
        type: 'Point',
        coordinates: [
          -76.9897,
          38.89011
        ]
      },
      properties: {
        label: 'Abraham Lincoln/Emancipation Monument, Washington, USA'
      }
    })

    // submit form
    tree.find('.commuter-submit-buttons').find('button').first().simulate('click')

    expectUpdateAction(mockStore.getActions())
  })

  it('Delete Commuter', () => {
    const mockStore = makeMockStore(mockStores.complexOrganization)
    window.confirm = () => true

    // Given a logged-in user is viewing the Create/Edit Commuter View
    // mount component
    const tree = mount(
      <Provider store={mockStore}>
        <EditCommuter
          params={{organizationId: '2', groupId: '1', commuterId: '1'}}
          />
      </Provider>
    , {
      attachTo: document.getElementById('test')
    })

    // When the user clicks the delete button
    // And the user confirms the Confirm Deletion dialog
    tree.find('.commuter-submit-buttons').find('button').last().simulate('click')

    expectDeleteCommuter(mockStore.getActions())
  })
})
