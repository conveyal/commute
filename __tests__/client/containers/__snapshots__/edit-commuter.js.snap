exports[`Container > EditCommuter Create commuter 1`] = `
Object {
  "payload": Object {
    "commuter": Object {
      "address": "Abraham Lincoln/Emancipation Monument, Washington, USA",
      "email": "mock@email.fake",
      "lat": 38.89011,
      "lng": -76.9897,
      "name": "Mock Commuter",
    },
    "groupId": "1",
    "organizationId": "2",
  },
  "type": "add commuter",
}
`;

exports[`Container > EditCommuter Create commuter 2`] = `
Object {
  "payload": Object {
    "args": Array [
      "/organizations/2/groups/1",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`Container > EditCommuter Create/Edit Commuter View loads in create mode 1`] = `
<div
  className="commuter-header col-xs-12">
  <h3>
    <span>
      Create Commuter
    </span>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="pull-right"
      disabled={false}>
      <button
        className="pull-right btn btn-default"
        disabled={false}
        type="button">
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/organizations/2/groups/1">
          <a
            onClick={[Function]}
            style={Object {}}>
            <Icon
              type="arrow-left">
              <i
                className="fa fa-arrow-left fa-fw " />
            </Icon>
            <span>
              Back
            </span>
          </a>
        </Link>
      </button>
    </Button>
  </h3>
</div>
`;

exports[`Container > EditCommuter Create/Edit Commuter View loads in create mode 2`] = `
<div
  className="commuter-form col-md-5 col-xs-12">
  <form>
    <FieldGroup
      label="Name"
      name="name"
      onChange={[Function]}
      placeholder="Enter name"
      type="text"
      value={undefined}>
      <FormGroup
        bsClass="form-group"
        controlId="group-item-name">
        <div
          className="form-group">
          <ControlLabel
            bsClass="control-label"
            srOnly={false}>
            <label
              className="control-label"
              htmlFor="group-item-name">
              Name
            </label>
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            placeholder="Enter name"
            type="text"
            value="">
            <input
              className="form-control"
              id="group-item-name"
              onChange={[Function]}
              placeholder="Enter name"
              type="text"
              value="" />
          </FormControl>
        </div>
      </FormGroup>
    </FieldGroup>
    <FieldGroup
      label="Email"
      name="email"
      onChange={[Function]}
      placeholder="Enter email"
      type="text"
      value={undefined}>
      <FormGroup
        bsClass="form-group"
        controlId="group-item-email">
        <div
          className="form-group">
          <ControlLabel
            bsClass="control-label"
            srOnly={false}>
            <label
              className="control-label"
              htmlFor="group-item-email">
              Email
            </label>
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            placeholder="Enter email"
            type="text"
            value="">
            <input
              className="form-control"
              id="group-item-email"
              onChange={[Function]}
              placeholder="Enter email"
              type="text"
              value="" />
          </FormControl>
        </div>
      </FormGroup>
    </FieldGroup>
    <Geocoder
      label="address"
      onChange={[Function]}
      value={undefined}>
      <FormGroup
        bsClass="form-group"
        controlId="geocode-control">
        <div
          className="form-group">
          <ControlLabel
            bsClass="control-label"
            srOnly={false}>
            <label
              className="control-label"
              htmlFor="geocode-control">
              address
            </label>
          </ControlLabel>
          <Geocoder
            apiKey="test-search-key"
            boundary={
              Object {
                "circle": Object {
                  "latlng": Object {
                    "lat": 38.8886,
                    "lng": -77.043,
                    "radius": 275,
                  },
                  "radius": 275,
                },
              }
            }
            featureToLabel={[Function]}
            featureToValue={[Function]}
            onChange={[Function]}
            search={[Function]}
            value={undefined}>
            <Async
              apiKey="test-search-key"
              autoload={false}
              boundary={
                Object {
                  "circle": Object {
                    "latlng": Object {
                      "lat": 38.8886,
                      "lng": -77.043,
                      "radius": 275,
                    },
                    "radius": 275,
                  },
                }
              }
              cache={Object {}}
              cacheAsyncResults={false}
              featureToLabel={[Function]}
              featureToValue={[Function]}
              filterOptions={false}
              ignoreAccents={true}
              ignoreCase={true}
              loadOptions={[Function]}
              loadingPlaceholder="Loading..."
              minimumInput={3}
              onChange={[Function]}
              options={Array []}
              search={[Function]}
              searchPromptText="Type to search"
              value={null}>
              <Select
                addLabelText="Add \"{label}\"?"
                apiKey="test-search-key"
                arrowRenderer={[Function]}
                autoload={false}
                autosize={true}
                backspaceRemoves={true}
                backspaceToRemoveMessage="Press backspace to remove {label}"
                boundary={
                  Object {
                    "circle": Object {
                      "latlng": Object {
                        "lat": 38.8886,
                        "lng": -77.043,
                        "radius": 275,
                      },
                      "radius": 275,
                    },
                  }
                }
                cache={Object {}}
                cacheAsyncResults={false}
                clearAllText="Clear all"
                clearValueText="Clear value"
                clearable={true}
                delimiter=","
                disabled={false}
                escapeClearsValue={true}
                featureToLabel={[Function]}
                featureToValue={[Function]}
                filterOptions={false}
                ignoreAccents={true}
                ignoreCase={true}
                inputProps={Object {}}
                isLoading={false}
                joinValues={false}
                labelKey="label"
                loadOptions={[Function]}
                loadingPlaceholder="Loading..."
                matchPos="any"
                matchProp="any"
                menuBuffer={0}
                menuRenderer={[Function]}
                minimumInput={3}
                multi={false}
                noResultsText="Type to search"
                onBlurResetsInput={true}
                onChange={[Function]}
                onCloseResetsInput={true}
                onInputChange={[Function]}
                openAfterFocus={false}
                optionComponent={[Function]}
                options={Array []}
                pageSize={5}
                placeholder="Select..."
                required={false}
                scrollMenuIntoView={true}
                search={[Function]}
                searchPromptText="Type to search"
                searchable={true}
                simpleValue={false}
                tabSelectsValue={true}
                value={null}
                valueComponent={[Function]}
                valueKey="value">
                <div
                  className="Select Select--single is-searchable"
                  style={undefined}>
                  <div
                    className="Select-control"
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    style={undefined}>
                    <span
                      className="Select-multi-value-wrapper"
                      id="react-select-3--value">
                      <div
                        className="Select-placeholder">
                        Select...
                      </div>
                      <AutosizeInput
                        aria-activedescendant="react-select-3--value"
                        aria-expanded="false"
                        aria-haspopup="false"
                        aria-label={undefined}
                        aria-labelledby={undefined}
                        aria-owns=""
                        className="Select-input"
                        minWidth="5px"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        required={false}
                        role="combobox"
                        tabIndex={undefined}
                        value="">
                        <div
                          className="Select-input"
                          style={
                            Object {
                              "display": "inline-block",
                            }
                          }>
                          <input
                            aria-activedescendant="react-select-3--value"
                            aria-expanded="false"
                            aria-haspopup="false"
                            aria-label={undefined}
                            aria-labelledby={undefined}
                            aria-owns=""
                            className={undefined}
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            required={false}
                            role="combobox"
                            style={
                              Object {
                                "boxSizing": "content-box",
                                "width": "5pxpx",
                              }
                            }
                            tabIndex={undefined}
                            value="" />
                          <div
                            style={
                              Object {
                                "height": 0,
                                "left": 0,
                                "overflow": "scroll",
                                "position": "absolute",
                                "top": 0,
                                "visibility": "hidden",
                                "whiteSpace": "pre",
                              }
                            } />
                        </div>
                      </AutosizeInput>
                    </span>
                    <span
                      className="Select-arrow-zone"
                      onMouseDown={[Function]}>
                      <span
                        className="Select-arrow"
                        onMouseDown={[Function]} />
                    </span>
                  </div>
                </div>
              </Select>
            </Async>
          </Geocoder>
        </div>
      </FormGroup>
    </Geocoder>
  </form>
</div>
`;

exports[`Container > EditCommuter Create/Edit Commuter View loads in create mode 3`] = `
<div
  className="commuter-submit-buttons col-xs-12">
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="success"
    disabled={false}
    onClick={[Function]}>
    <button
      className="btn btn-success"
      disabled={false}
      onClick={[Function]}
      type="button">
      Create
    </button>
  </Button>
</div>
`;

exports[`Container > EditCommuter Create/Edit Commuter View loads in edit mode 1`] = `
<div
  className="commuter-header col-xs-12">
  <h3>
    <span>
      Edit Commuter
    </span>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="pull-right"
      disabled={false}>
      <button
        className="pull-right btn btn-default"
        disabled={false}
        type="button">
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/organizations/2/groups/1">
          <a
            onClick={[Function]}
            style={Object {}}>
            <Icon
              type="arrow-left">
              <i
                className="fa fa-arrow-left fa-fw " />
            </Icon>
            <span>
              Back
            </span>
          </a>
        </Link>
      </button>
    </Button>
  </h3>
</div>
`;

exports[`Container > EditCommuter Create/Edit Commuter View loads in edit mode 2`] = `
<div
  className="commuter-form col-md-5 col-xs-12">
  <form>
    <FieldGroup
      label="Name"
      name="name"
      onChange={[Function]}
      placeholder="Enter name"
      type="text"
      value="Luke Warm">
      <FormGroup
        bsClass="form-group"
        controlId="group-item-name">
        <div
          className="form-group">
          <ControlLabel
            bsClass="control-label"
            srOnly={false}>
            <label
              className="control-label"
              htmlFor="group-item-name">
              Name
            </label>
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            placeholder="Enter name"
            type="text"
            value="Luke Warm">
            <input
              className="form-control"
              id="group-item-name"
              onChange={[Function]}
              placeholder="Enter name"
              type="text"
              value="Luke Warm" />
          </FormControl>
        </div>
      </FormGroup>
    </FieldGroup>
    <FieldGroup
      label="Email"
      name="email"
      onChange={[Function]}
      placeholder="Enter email"
      type="text"
      value="luke@warm.cold">
      <FormGroup
        bsClass="form-group"
        controlId="group-item-email">
        <div
          className="form-group">
          <ControlLabel
            bsClass="control-label"
            srOnly={false}>
            <label
              className="control-label"
              htmlFor="group-item-email">
              Email
            </label>
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            onChange={[Function]}
            placeholder="Enter email"
            type="text"
            value="luke@warm.cold">
            <input
              className="form-control"
              id="group-item-email"
              onChange={[Function]}
              placeholder="Enter email"
              type="text"
              value="luke@warm.cold" />
          </FormControl>
        </div>
      </FormGroup>
    </FieldGroup>
    <Geocoder
      label="address"
      onChange={[Function]}
      value={
        Object {
          "label": "4321 XYZ Boulevard",
        }
      }>
      <FormGroup
        bsClass="form-group"
        controlId="geocode-control">
        <div
          className="form-group">
          <ControlLabel
            bsClass="control-label"
            srOnly={false}>
            <label
              className="control-label"
              htmlFor="geocode-control">
              address
            </label>
          </ControlLabel>
          <Geocoder
            apiKey="test-search-key"
            boundary={
              Object {
                "circle": Object {
                  "latlng": Object {
                    "lat": 38.8886,
                    "lng": -77.043,
                    "radius": 275,
                  },
                  "radius": 275,
                },
              }
            }
            featureToLabel={[Function]}
            featureToValue={[Function]}
            onChange={[Function]}
            search={[Function]}
            value={
              Object {
                "label": "4321 XYZ Boulevard",
              }
            }>
            <Async
              apiKey="test-search-key"
              autoload={false}
              boundary={
                Object {
                  "circle": Object {
                    "latlng": Object {
                      "lat": 38.8886,
                      "lng": -77.043,
                      "radius": 275,
                    },
                    "radius": 275,
                  },
                }
              }
              cache={Object {}}
              cacheAsyncResults={false}
              featureToLabel={[Function]}
              featureToValue={[Function]}
              filterOptions={false}
              ignoreAccents={true}
              ignoreCase={true}
              loadOptions={[Function]}
              loadingPlaceholder="Loading..."
              minimumInput={3}
              onChange={[Function]}
              options={Array []}
              search={[Function]}
              searchPromptText="Type to search"
              value={
                Object {
                  "label": "4321 XYZ Boulevard",
                }
              }>
              <Select
                addLabelText="Add \"{label}\"?"
                apiKey="test-search-key"
                arrowRenderer={[Function]}
                autoload={false}
                autosize={true}
                backspaceRemoves={true}
                backspaceToRemoveMessage="Press backspace to remove {label}"
                boundary={
                  Object {
                    "circle": Object {
                      "latlng": Object {
                        "lat": 38.8886,
                        "lng": -77.043,
                        "radius": 275,
                      },
                      "radius": 275,
                    },
                  }
                }
                cache={Object {}}
                cacheAsyncResults={false}
                clearAllText="Clear all"
                clearValueText="Clear value"
                clearable={true}
                delimiter=","
                disabled={false}
                escapeClearsValue={true}
                featureToLabel={[Function]}
                featureToValue={[Function]}
                filterOptions={false}
                ignoreAccents={true}
                ignoreCase={true}
                inputProps={Object {}}
                isLoading={false}
                joinValues={false}
                labelKey="label"
                loadOptions={[Function]}
                loadingPlaceholder="Loading..."
                matchPos="any"
                matchProp="any"
                menuBuffer={0}
                menuRenderer={[Function]}
                minimumInput={3}
                multi={false}
                noResultsText="Type to search"
                onBlurResetsInput={true}
                onChange={[Function]}
                onCloseResetsInput={true}
                onInputChange={[Function]}
                openAfterFocus={false}
                optionComponent={[Function]}
                options={Array []}
                pageSize={5}
                placeholder="Select..."
                required={false}
                scrollMenuIntoView={true}
                search={[Function]}
                searchPromptText="Type to search"
                searchable={true}
                simpleValue={false}
                tabSelectsValue={true}
                value={
                  Object {
                    "label": "4321 XYZ Boulevard",
                  }
                }
                valueComponent={[Function]}
                valueKey="value">
                <div
                  className="Select Select--single is-searchable has-value"
                  style={undefined}>
                  <div
                    className="Select-control"
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    style={undefined}>
                    <span
                      className="Select-multi-value-wrapper"
                      id="react-select-4--value">
                      <Value
                        disabled={false}
                        id="react-select-4--value-item"
                        instancePrefix="react-select-4-"
                        onClick={null}
                        value={
                          Object {
                            "label": "4321 XYZ Boulevard",
                          }
                        }>
                        <div
                          className="Select-value"
                          style={undefined}
                          title={undefined}>
                          <span
                            aria-selected="true"
                            className="Select-value-label"
                            id="react-select-4--value-item"
                            role="option">
                            4321 XYZ Boulevard
                          </span>
                        </div>
                      </Value>
                      <AutosizeInput
                        aria-activedescendant="react-select-4--value"
                        aria-expanded="false"
                        aria-haspopup="false"
                        aria-label={undefined}
                        aria-labelledby={undefined}
                        aria-owns=""
                        className="Select-input"
                        minWidth="5px"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        required={false}
                        role="combobox"
                        tabIndex={undefined}
                        value="">
                        <div
                          className="Select-input"
                          style={
                            Object {
                              "display": "inline-block",
                            }
                          }>
                          <input
                            aria-activedescendant="react-select-4--value"
                            aria-expanded="false"
                            aria-haspopup="false"
                            aria-label={undefined}
                            aria-labelledby={undefined}
                            aria-owns=""
                            className={undefined}
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            required={false}
                            role="combobox"
                            style={
                              Object {
                                "boxSizing": "content-box",
                                "width": "5pxpx",
                              }
                            }
                            tabIndex={undefined}
                            value="" />
                          <div
                            style={
                              Object {
                                "height": 0,
                                "left": 0,
                                "overflow": "scroll",
                                "position": "absolute",
                                "top": 0,
                                "visibility": "hidden",
                                "whiteSpace": "pre",
                              }
                            } />
                        </div>
                      </AutosizeInput>
                    </span>
                    <span
                      aria-label="Clear value"
                      className="Select-clear-zone"
                      onMouseDown={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      title="Clear value">
                      <span
                        className="Select-clear"
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&times;",
                          }
                        } />
                    </span>
                    <span
                      className="Select-arrow-zone"
                      onMouseDown={[Function]}>
                      <span
                        className="Select-arrow"
                        onMouseDown={[Function]} />
                    </span>
                  </div>
                </div>
              </Select>
            </Async>
          </Geocoder>
        </div>
      </FormGroup>
    </Geocoder>
  </form>
</div>
`;

exports[`Container > EditCommuter Create/Edit Commuter View loads in edit mode 3`] = `
<div
  className="commuter-submit-buttons col-xs-12">
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="warning"
    disabled={false}
    onClick={[Function]}>
    <button
      className="btn btn-warning"
      disabled={false}
      onClick={[Function]}
      type="button">
      Update
    </button>
  </Button>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="danger"
    disabled={false}
    onClick={[Function]}>
    <button
      className="btn btn-danger"
      disabled={false}
      onClick={[Function]}
      type="button">
      Delete
    </button>
  </Button>
</div>
`;

exports[`Container > EditCommuter Create/Edit Commuter View loads in edit mode 4`] = `
Object {
  "lat": 38.916089,
  "lng": -76.970221,
  "lon": -76.970221,
}
`;

exports[`Container > EditCommuter Delete Commuter 1`] = `
Array [
  Object {
    "payload": Object {
      "commuterId": "1",
      "groupId": "1",
      "organizationId": "2",
    },
    "type": "delete commuter",
  },
  Object {
    "payload": Object {
      "args": Array [
        "/organizations/2/groups/1",
      ],
      "method": "push",
    },
    "type": "@@router/CALL_HISTORY_METHOD",
  },
]
`;

exports[`Container > EditCommuter Update commuter 1`] = `
Object {
  "payload": Object {
    "commuter": Object {
      "address": "Abraham Lincoln/Emancipation Monument, Washington, USA",
      "email": "mock@email.fake",
      "id": "1",
      "lat": 38.89011,
      "lng": -76.9897,
      "name": "Mock Commuter",
    },
    "groupId": "1",
    "organizationId": "2",
  },
  "type": "update commuter",
}
`;

exports[`Container > EditCommuter Update commuter 2`] = `
Object {
  "payload": Object {
    "args": Array [
      "/organizations/2/groups/1",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;
